source/server/config/http/lightstep_http_tracer.cc

```diff
@@ -0,0 +1,44 @@
+#include "server/config/http/lightstep_http_tracer.h"
+
+#include <string>
+
+#include "common/tracing/http_tracer_impl.h"
+#include "common/tracing/lightstep_tracer_impl.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+Tracing::HttpTracerPtr LightstepTracerFactory::tryCreateHttpTracer(const std::string &type,
+                                                const Json::Object& json_config,
+                                                Server::Instance& server,
+                                                Upstream::ClusterManager &cluster_manager) {
+  if (type != "lightstep") {
```

Comment:
I was kind of hoping for an approach more like: map<string, factory>
Then we don't rely on the factory to decide if it likes the type+config, we just choose the factory by name, and pass it the config. However, I take it that this is more inline with existing Envoy practice?